src/domain/books.rs

```diff
@@ -0,0 +1,56 @@
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Book {
+    pub book_id: String,
+    pub author_id: Option<Uuid>,
+    pub collection_id: Option<Uuid>,
+    pub name: String,
+    pub genre: Genre,
+    pub lang: Lang,
+    pub rating: f64,
+    pub creation_time: DateTime<Utc>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub deletion_time: Option<DateTime<Utc>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub updated_at: Option<DateTime<Utc>>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateBook {
+    pub name: String,
+    pub author_id: Uuid,
+    pub collection_id: Uuid,
```

Comment:
Nem sempre o livro vai estar em uma coleção, acho que aqui é:

```suggestion
pub struct CreateBook {
    pub name: String,
    pub author_id: Uuid,
    pub collection_id: Option<Uuid>,
```